### PR TITLE
Add sqlite support

### PR DESCRIPTION
**Pull request by daveworth**
This PR adds SQLite support to the package since it uses a compatible savepoint syntax to Postgres.   I am starting to experiment with running SQLite as an "in-memory" store for tests and development and Postgres for production and adding this level of compatibility feels pretty good.  Please let me know if there is more you'd like before merging this request - I'm more than happy to do so.

### CODE REVIEW

**Comment by skateinmars** [comment]
Hello @daveworth, apologies for the extra late reply

We recently added CI integration (cf https://github.com/heetch/sqalx/pull/19), would you mind rebasing your PR to use the Makefile?
Thanks in advance!

**Comment by daveworth** [comment]
Hi @skateinmars - I've rebased and pushed.  We should be set to go.  I'm looking forward to any additional changes you'd like to this PR.

File: README.md [hunk 1]
```diff
@@ -104,5 +104,32 @@ node, err := sqalx.New(db, sqalx.SavePoint(true))
 ## Issue
 Please open an issue if you encounter any problem.
 
+## Development
+sqalx is covered by a go test suite.  In order to test against specific databases we include a docker-compose file that runs Postgres and MySQL.
+
+### Running all tests
+To run the tests, first run `docker-compose up` to run both Postgres and MySQL in locally-exposed docker iamges.  The run your tests via `./test.sh` which sets up the above described data sources and runs `go test -v ./...`.
```
Comment (skateinmars) [inline README.md:111]:
```suggestion
To run the tests, first run `docker-compose up` to run both Postgres and MySQL in locally-exposed docker images.  Then run your tests via `make test` which sets up the above described data sources and runs `go test -v ./...`.
```

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+version: '3.6'
+
+services:
+    postgres:
+        image: postgres:9.6-alpine
+        ports:
+            - 5432:5432
+        environment:
+            # POSTGRESQL_DATASOURCE="postgresql://sqalx:sqalx@localhost:5432/sqalx?sslmode=disable"
```
Comment (skateinmars) [inline docker-compose.yml:9]:
```suggestion

```

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            - POSTGRES_USER=sqalx
+            - POSTGRES_PASSWORD=sqalx
+
+    mysql:
+        image: mysql:8.0 # intel only
+        # image: mysql/mysql-server:8.0 # mac M1 preview
+        ports:
+            - 3306:3306
+        environment:
+            # MYSQL_DATASOURCE="sqalx:sqalx@tcp(localhost:3306)/sqalx"
```
Comment (skateinmars) [inline docker-compose.yml:19]:
```suggestion
```

File: test.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (skateinmars) [inline test.sh:1]: We can remove this file to use the Makefile instead (the Makefile will just need to be updated to use `SQLITE_DATASOURCE`

**Review by skateinmars — COMMENTED** [review]
Thanks for this contribution! Just a couple comments about the use of the Makefile

File: README.md (hunk 1 above)
Comment (daveworth) [reply README.md:111]: Oof - good catch.  I'm pretty embarrassed by those typos.

**Comment by daveworth** [comment]
Suggested changes made (thank you!  This is much better now!)

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        image: postgres:9.6-alpine
+        ports:
+            - 5432:5432
+        environment:
+            - POSTGRES_USER=sqalx
+            - POSTGRES_PASSWORD=sqalx
+
+    mysql:
+        # image: mysql:8.0 # intel only
+        image: mysql/mysql-server:8.0 # mac M1 preview
```
Comment (sixstone-qq) [inline docker-compose.yml:14]: I'd do it the other way around unless this image works in all platforms.

File: sqalx.go [hunk 6]
```diff
@@ -215,7 +215,8 @@ type Option func(*node) error
 // SavePoint option enables PostgreSQL Savepoints for nested transactions.
```
Comment (sixstone-qq) [inline sqalx.go:215]: This comment should be updated if SQLite supports SavePoints

**Review by sixstone-qq — COMMENTED** [review]
Two minor comments, thanks a lot for the contribution! :hugs:

File: docker-compose.yml (hunk 5 above)
Comment (daveworth) [reply docker-compose.yml:14]: yes, sorry - artifact of me testing on my Mac M1 :-D

File: sqalx.go (hunk 6 above)
Comment (daveworth) [reply sqalx.go:215]: good catch - thank you.

**Comment by skateinmars** [comment]
Merging, thanks again for your contribution!

**Comment by sixstone-qq** [comment]
Thanks a lot @daveworth for this! :hugs:

**Pull request merged**